airflow/secrets/local_filesystem.py

```diff
@@ -0,0 +1,257 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Objects relating to retreiving connections and variables from local file
+"""
+import json
+import logging
+import os
+from collections import defaultdict
+from json import JSONDecodeError
+from typing import Any, Dict, List, Optional, Tuple
+
+from airflow.exceptions import AirflowException, AirflowFileParseException, FileSyntaxError
+from airflow.secrets.base_secrets import BaseSecretsBackend
+from airflow.utils.file import COMMENT_PATTERN
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+# constants to limit cyclical imports
+CONNECTION_PARAMETERS_NAMES = {
+    "conn_id",
+    "conn_type",
+    "host",
+    "login",
+    "password",
+    "schema",
+    "port",
+    "extra",
+    "uri",
+}
+
+log = logging.getLogger(__name__)
+
+
+def _parser_env_file(content: str) -> Tuple[Dict[str, List[str]], List[FileSyntaxError]]:
```

Comment:
If the name is `_parser_env_file`, how about we take the file as an input?

Reply:
I want to refactor CLI and Web UI to use the same code. I want to refactor CLI and Web UI to use the same code. in the next PR. For Web UI, it would be useful for the user to paste the contents of the file instead of uploading the file.

Reply:
The Variables uploaded from the UI are saved to DB. Where will this code be reused?

Reply:
I would like to add support for env files and better error handling 

Please look at this code:
https://github.com/apache/airflow/blob/master/airflow/cli/commands/variable_command.py#L76-L79
A pure exception is caught here. The error message is very inaccurate. This is not user friendly.

After that, the code in Web UI and CLI is duplicated. Now it will be the third-place where the logic responsible for loading variables from the file is.

Reply:
<img width="655" alt="Screenshot 2020-04-18 at 21 24 12" src="https://user-images.githubusercontent.com/12058428/79669319-ff93d500-81ba-11ea-8c71-95c15ad50f0f.png">